src/XrdCeph/XrdCephPosix.cc

```diff
@@ -417,14 +419,16 @@ void fillCephFile(const char *path, XrdOucEnv *env, CephFile &file) {
   // If env is null or no entry is found for what is missing, defaults are
   // applied. These defaults are initially set to 'admin', 'default', 1, 4MB and 4MB
   // but can be changed via a call to ceph_posix_set_defaults
-  std::string spath = path;
+  std::string spath {path};
+  // If namelib is specified, apply translation to the whole path (which might include pool, etc)
+  translateFileName(spath,path);
```

Comment:
would the translation affect colonPos? in previous version colonPos is calculated before the translation

Reply:
This is the main motivation for the fix. The name mapping to be applied to the whole "pathname", e.g. it should consider "/dteam:myfile", not just "myfile". In our case the storage.xml file removes the extraneous slash at the start of the path. This is potentially a breaking change for any other site using this code, and with an obscure mapping. But I don't believe this is the case.